test_problems/clib_test/clib_test.c

```diff
@@ -15,7 +15,7 @@ int main(int argc, char** argv)
 {
     int ret;
     int thermo = thermo_newFromFile("gri30.yaml", "gri30");
```

Comment:
Thanks for catching this: any idea why this was not raising issues? I had thought of removing those checks as they are superseded by the new test suite (checks are replicated), but held off in #1448.

Reply:
I was confused by this for a while, too. The answer is that `assert` statements all get optimized out when compiling with the `NDEBUG` macro enabled, which we generally do unless you specify `optimize=n` to SCons. I happened to catch this because I was debugging a different crash and needed to disable optimizations to be able to use the debugger effectively.